### ISSUE TITLE
Segmentation Fault when trying to run my own custom op with default EP.

### ISSUE THREAD

**Issue by Eralien** [microsoft/onnxruntime#4754]
**The bug behavior**
When I am trying to run my model with custom ops, a segmentation fault occurs inside my custom op kernel. The error occurs when calling `CustomOpApi::KernelContext_GetInput`, which fails to be initialized and so does many other callback functions within the private member `OrtApi api_` of my custom op kernel.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- ONNX Runtime installed from (source or binary): source
- ONNX Runtime version: 1.4.0
- GCC/Compiler version (if compiling from source): 8.4
- CUDA/cuDNN version: 10.2

**To Reproduce**
This behavior happened when I was using my own onnx exported network. My code was strictly following the gtest `onnxruntime_shared_lib_test::CApiTest::test_custom_library` in `TestInference.cc` as instructed in documentation.
Basically my practice was first to replicate what test_custom_library did. Then I replaced the custom op kernel, the inputs and other stuff with my impl. Then this error occurred.  

**Expected behavior**
I would expect all callback functions inside the OrtApi to be properly initialized like the gtest example.

**Screenshots**
Two screenshots on the last and second to last frame where the error occurred.

1. `api.KernelContext_GetInput` callback was not initialized.
![ort_cxx_inline:600](https://user-images.githubusercontent.com/26458167/89881050-90be0a00-dbf7-11ea-82fd-0cb144a86583.png)
2. Corresponding location in my own kernel compute impl.
![custom_op:45](https://user-images.githubusercontent.com/26458167/89881335-f6aa9180-dbf7-11ea-89d3-0802a64d0379.png)

**Comment by hariharans29** [comment]
CC: @RyanUnderhill

**Comment by Eralien** [comment]
Hi there, I still unable to identify the reason for the bug. Is there any hint please?

**Comment by RyanUnderhill** [comment]
@Eralien Ah, there's a bug in the sample, OrtApi is meant to just be used as a pointer, not to be copied by value. Is it uninitialized when passed to your constructor? (hard to tell through the portion of code shared).

**Comment by j-suyako** [comment]
same problem, you should use **api_(not api!)** when initialize kernel, in this case:
`KernelFurthestPointSampling(OrtApi api): api_(api), ort_(api_){}`
